workers/unity/Packages/com.improbable.gdk.core/Tests/Editmode/Utility/EntityBuilderTest.cs

```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using NUnit.Framework;
+using Improbable.Worker.Core;
+using UnityEngine;
+
+namespace Improbable.Gdk.Core.EditmodeTests
+{
+    [TestFixture]
+    public class EntityBuilderTest
+    {
+        [Test]
+        public void EntityBuilder_should_throw_exception_if_no_position_component_is_added()
+        {
+            var builder = EntityBuilder.Begin();
+            Assert.Throws<InvalidEntityException>(() => builder.Build());
+        }
+
+        [Test]
+        public void EntityBuilder_should_throw_exception_if_component_is_added_twice()
+        {
+            var builder = EntityBuilder.Begin();
+            builder.AddPosition(0, 0, 0, "write-access");
+            Assert.Throws<InvalidOperationException>(() => builder.AddPosition(0, 0, 0, "write-access"));
+
+            DisposeEntity(builder.Build());
+        }
+
+        [Test]
+        public void EntityBuilder_should_throw_exception_if_Build_called_more_than_once()
+        {
+            var builder = EntityBuilder.Begin();
+            builder.AddPosition(0, 0, 0, "write-access");
+            var entity = builder.Build();
+            Assert.Throws<InvalidOperationException>(() => builder.Build());
+
+            DisposeEntity(entity);
+        }
+
+        [Test]
+        public void EntityBuilder_should_Build_if_only_position_is_added()
+        {
+            var builder = EntityBuilder.Begin();
+            builder.AddPosition(0, 0, 0, "write-access");
+
+            Entity entity = new Entity(); // Stops IDE complaining about null check below.
+            Assert.DoesNotThrow(() => entity = builder.Build());
+
+            DisposeEntity(entity);
+        }
+
+        [Test]
+        public void EntityBuilder_should_Build_if_arbitraty_components_are_added()
+        {
+            var builder = EntityBuilder.Begin();
+            builder.AddPosition(0, 0, 0, "write-access");
+            builder.AddComponent(GetComponentDataWithId(1000), "write-access");
+            builder.AddComponent(GetComponentDataWithId(1001), "write-access");
+
+            Entity entity = new Entity(); // Stops IDE complaining about null check below.
+            Assert.DoesNotThrow(() => entity = builder.Build());
+
+            DisposeEntity(entity);
```

Comment:
put this within a `finally` in case the assertion above may fail (may cause a snowball test failure effect to haunt us)

Reply:
Similarly for other `DisposeEntity` calls